core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnInsertExec.scala

```diff
@@ -156,7 +157,8 @@ case class ColumnInsertExec(_child: SparkPlan, partitionColumns: Seq[String],
        |  int $defaultRowSize = 0;
        |  ${declarations.mkString("\n")}
        |  $defaultBatchSizeTerm = Math.max(
-       |    (${math.abs(columnBatchSize)} - 8) / $defaultRowSize, 16);
+       |    (${math.abs( if (columnBatchSize > 0) columnBatchSize else Property.ColumnBatchSize.
+            defaultValue.get)} - 8) / $defaultRowSize, 16);
```

Comment:
Had a relook at your comments and code and this is what I think:

Lets pass an additional variable defaultBatchSizeTerm right through ColumnFormatSamplingRelation to ColumnInsertExec.